src/components/IndexTable/components/Row/tests/Row.test.tsx

```diff
@@ -209,6 +209,30 @@ describe('<Row />', () => {
     expect(onSelectionChangeSpy).toHaveBeenCalledTimes(1);
   });
 
+  it('does not fire onClick handler when row is clicked and no primary link child present and table is not selectable', () => {
```

Comment:
[suggestion] could also add tests for `checkbox is shown when selectable is true` and `not shown when selectable is false`